src/extension/inpage-script/index.js

```diff
@@ -121,6 +121,21 @@ if (document) {
     },
     { capture: true }
   );
+  // Listen for webln events from the extension
+  // emit events to the websites
+  window.addEventListener("message", (event) => {
+    if (event.source === window && event.data.action === "accountChanged") {
+      eventEmitter(event.data.action, event.data.scope);
+    }
+  });
 } else {
   console.warn("Failed to inject WebLN provider");
 }
+
+function eventEmitter(action, scope) {
+  if (window[scope] && window[scope]._eventEmitter) {
+    window[scope].emit(action);
+  } else {
```

Comment:
`else return` is redundant?

Why check for a private variable `_eventEmitter` but then call `.emit`? maybe you can check  for `.emit`?

Can action and scope variables be typed?

Reply:
why typing is needed here scope will always be passed as string as it's not on user end and on extension side